### PR TITLE
feat!: Add EnableNameFieldEscape config

### PR DESCRIPTION
**Pull request by weichou1229**
BREAKING CHANGE: Add EnableNameFieldEscape config on the service level and allow NameFieldEscape configurable

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) not impact
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) not impact
  <link to docs PR>

## Testing Instructions

Related PRs are opened, you can refer to https://github.com/edgexfoundry/edgex-go/issues/4673#issuecomment-1709699336 and run services to verify the API.

### CODE REVIEW

File: config/types.go [hunk 1]
```diff
@@ -62,6 +62,9 @@ type ServiceInfo struct {
 	// RequestTimeout specifies a timeout (in milliseconds) for
 	// processing REST request calls from other services.
 	RequestTimeout string
+	// EnableNameFieldEscape indicates whether enables NameFieldEscape in this service
+	// The name field escape could allow the system to use special or Chinese characters in the different name fields, including device, profile, and so on.  If the EnableNameFieldEscape is false, some special characters might cause system error.
+	EnableNameFieldEscape bool
```
Comment (lenny-intel) [inline config/types.go:67]: Please add TODO comment to remove in EdgeX 4.0

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: config/types.go (hunk 1 above)
Comment (weichou1229) [reply config/types.go:68]: Added comment.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by weichou1229** [comment]
recheck

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**